### DISCUSSION TITLE
Proper remote IP detection behind Cloudflare or the like

### DISCUSSION THREAD

**Discussion by t7tran** [directus/directus#10069] [Feature Requests]
We have instances proxied through Cloudflare and all IP addresses captured for Activities are from Cloudflare. It likely affects the `check-ip` middleware too. How can we correctly detect and capture the actual client IPs?

I looked at the code to see `req.ip` being used everywhere from `app.ts`, `server.ts`, to auth drivers implementation, even `graphql.ts`. Should we replace all `req.ip`/`req?.ip` occurrences with `req.real_ip` which is populated by a custom middleware (similar to this plugin https://github.com/Mewte/cloudflare-express)?

**Comment by rijkvanzanten** [comment]
Very good question. I'm wondering how we can make this configurable. It seems like most CDNs will use a (custom) header that contains this information, in this case `cf-connecting-ip`. We might be able to support this properly by adding a `IP_EXTRACT_FROM_HEADER` (or similar) environment variable to control it, as that would also allow it to be used with other CDNs (and it could default to `X-Forwarded-For`)

Update: Implemented in #10765.
